src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -129,33 +154,52 @@ private OnnxTransform(IHostEnvironment env, Arguments args, byte[] modelBytes =
                 Model = OnnxModel.CreateFromBytes(modelBytes);
 
             var modelInfo = Model.ModelInfo;
-            if (modelInfo.InputsInfo.Length != 1)
-                throw env.Except($"OnnxTransform supports Onnx models with one input. The provided model has ${modelInfo.InputsInfo.Length} input(s).");
-            if (modelInfo.OutputsInfo.Length != 1)
-                throw env.Except($"OnnxTransform supports Onnx models with one output. The provided model has ${modelInfo.OutputsInfo.Length} output(s).");
-
-            Input = args.InputColumn;
-            Output = args.OutputColumn;
-
-            var outputNodeInfo = Model.ModelInfo.OutputsInfo[0];
-            var type = OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type);
-            var shape = outputNodeInfo.Shape;
-            var dims = shape.Count > 0 ? shape.Skip(shape[0] < 0 ? 1 : 0).Select( x => (int) x ).ToArray() : new[] { 0 };
-            OutputType = new VectorType(type, dims);
+            //if (modelInfo.InputsInfo.Length != 1)
+            //    throw env.Except($"OnnxTransform supports Onnx models with one input. The provided model has ${modelInfo.InputsInfo.Length} input(s).");
+            //if (modelInfo.OutputsInfo.Length != 1)
+            //    throw env.Except($"OnnxTransform supports Onnx models with one output. The provided model has ${modelInfo.OutputsInfo.Length} output(s).");
+
+            Inputs = args.InputColumns;
+            Outputs = args.OutputColumns;
+            //var type = OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type);
+            //var shape = outputNodeInfo.Shape;
+            //var dims = shape.Count > 0 ? shape.Skip(shape[0] < 0 ? 1 : 0).Select( x => (int) x ).ToArray() : new[] { 0 };
+
+            OutputTypes = new ColumnType[args.OutputColumns.Length];
+
+            var numModelOutputs = Model.ModelInfo.OutputsInfo.Length;
+            for (int i=0; i < args.OutputColumns.Length; i++)
+            {
+                var idx = -1;
+                for (var j = 0; j < Model.ModelInfo.OutputsInfo.Length; j++)
+                    if (Model.ModelInfo.OutputsInfo[j].Name == args.OutputColumns[i])
+                    {
+                        idx = j;
+                        break;
+                    }
+                if (idx < 0)
+                    throw _host.Except($"Column {args.OutputColumns[i]} doesn't match output node names of model");
```

Comment:
I would make OnnxModel._outputNames public and simplify code to:
if (!Model.OutputName.Contains(args.OutputColumns[i])
   throw ...

Reply:
simplified .

---
In reply to: [232835603](https://github.com/dotnet/machinelearning/pull/1586#discussion_r232835603) [](ancestors = 232835603)

Reply:
why not use Contains() instead of IndexOf?

---
In reply to: [234085225](https://github.com/dotnet/machinelearning/pull/1586#discussion_r234085225) [](ancestors = 234085225,232835603)

Reply:
The line of code just below uses the value -- we need to know the index since the user specified inputs might not be in the same order as Model.ModelInfo.OutputsInfo.

---
In reply to: [234305001](https://github.com/dotnet/machinelearning/pull/1586#discussion_r234305001) [](ancestors = 234305001,234085225,232835603)